include/sys/zio_compress.h

```diff
@@ -51,6 +51,7 @@ enum zio_compress {
 	ZIO_COMPRESS_GZIP_9,
 	ZIO_COMPRESS_ZLE,
 	ZIO_COMPRESS_LZ4,
+	ZIO_COMPRESS_EXTERNAL,
```

Comment:
The compression type is stored in every block pointer so adding a new one means changing the on-disk format, and adding a new feature flag for compatibility.  For example, see the `lz4_compress` feature flag in `zpool-features(5)`.

Before we do that I'd like to better understand why you ruled out using one of the existing compression types.  Why not `ZIO_COMPRESS_OFF` when it's known the data provided by Lustre is already compressed?

Reply:
I think the decision creating a new compression type was made some time ago before I started working on by bachelor thesis. Honestly I never questioned the decision, because I liked the idea of having data explicitly marked as pre-compressed. But I see the problem of a new compression type and the compatibility problem.

I tried using `ZIO_COMPRESS_OFF` and it worked after removing an assert statement in `arc_alloc_compressed_buf`, which I'm not happy with. After searching for other assert statements regarding `lsize` and `psize`, I discovered some of them. For example in `decode_embedded_bp` it says `ASSERT3U(lsize, ==, psize)` when `ZIO_COMPRESS_OFF` is used in the block pointer.

Actually I don't know if all (or even any) of them are relevant but that lets me think if it's the final idea to use `ZIO_COMPRESS_OFF`.